src/Uno.UWP/Storage/StorageFile.cs

```diff
@@ -33,126 +33,100 @@ private static Task<StorageFile> GetFileFromApplicationUri(CancellationToken ct,
 			=> throw new NotImplementedException();
 #endif
 
-		private readonly ImplementationBase _impl;
-
-		private StorageFile(ImplementationBase impl)
+		private StorageFile(ImplementationBase implementation)
 		{
-			_impl = impl;
-			_impl.InitOwner(this);
+			Implementation = implementation;
+			Implementation.InitOwner(this);
 		}
 
-		public string Path => _impl.Path;
+		internal ImplementationBase Implementation { get; }

```

Comment:
I would really prefer to keep it private. Why do we need to make it externally accessible? 

BTW I'm wondering if at some point this "implementation" should be provided by the `Provider` (I noticed the `Provider` property only after I've added those "implementation" stuff).

Reply:
I think I had a reason why I needed to make it `internal`, need to check. However, it really does make great sense to move it to `Provider`. If you don't mind, I would do it in some later PR though, as I have several branches based on this one and major changes would cause many merge conflicts 😀

Reply:
Hmm, take it back a bit - checking from UWP it seems that all files that `Provider` returns the exact same for all files and folder picked from the same storage (so you can compare whether two files come from the same "source". In our case however, the implementation base is file- or folder- specific (and has different operations for both), so that would probably be confusing. Especially as the implementation for the same operation may be different between folder and file for the same provider (deletion for example, must be recursive for directories)

Reply:
Interesting! So yes it's much more complex than just extracting files, we can definitively keep it for another PR